File: src/main/java/com/camackenzie/exvi/core/api/DataResult.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.camackenzie.exvi.core.api;

/**
 *
 * @author callum
 */
public class DataResult<T> {

    private final T result;
    private final int error;
    private final String message;

    public DataResult(int err, String msg, T res) {
        this.error = err;
        this.message = msg;
        this.result = res;
    }

    public T getResult() {
        return this.result;
    }

    public int getError() {
        return this.error;
    }

    public String getMessage() {
        return this.message;
    }

}

File: exvi-core/src/commonMain/kotlin/com/camackenzie/exvi/core/api/DataResult.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.camackenzie.exvi.core.api

import com.camackenzie.exvi.core.util.SelfSerializable

/**
 *
 * @author callum
 */
@kotlinx.serialization.Serializable
abstract class DataResult<T: SelfSerializable>(val error: Int, val message: String): SelfSerializable {
    abstract val result: T?

    fun errorOccured(): Boolean {
        return error != 0
    }
}